plugin-dev/Source/Sentry/Private/Desktop/SentryScopeDesktop.cpp

```diff
@@ -17,12 +17,26 @@ SentryScopeDesktop::SentryScopeDesktop()
 {
 }
 
+SentryScopeDesktop::SentryScopeDesktop(const SentryScopeDesktop& Scope)
+{
+	Dist = Scope.Dist;
+	Environment = Scope.Environment;
+	FingerprintDesktop = Scope.FingerprintDesktop;
+	TagsDesktop = Scope.TagsDesktop;
+	ExtraDesktop = Scope.ExtraDesktop;
+	ContextsDesktop = Scope.ContextsDesktop;
+	BreadcrumbsDesktop = Scope.BreadcrumbsDesktop;
+	LevelDesktop = Scope.LevelDesktop;
+}
+
```

Comment:
Where is this coming from?

Reply:
It's essential to implement a custom copy constructor for the scope class after we've introduced a critical section member.

Since `FCriticalSection` disables its default copy constructor using the default mechanism for scope isn't possible and results into compilation errors.

Reply:
TIL. Thanks!